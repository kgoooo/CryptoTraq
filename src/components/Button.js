import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';

const Button = (props) => {
	return (
		<button
			onClick={props.onClick}
			style={[styles.btn__main, props.buttonAnimation]}
		>{props.buttonContent}</button>
	)
};

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	buttonContent: PropTypes.string.isRequired,
	buttonAnimation: PropTypes.object
};

/*  Button styles for Radium defined here and added below to the component on the export statement. */
const styles = {
	btn__main: {
		background: 'transparent',
		borderRadius: '8px',
		border: `2px solid ${RadiumVars.color.colorPrimaryLightBlue}`,
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontSize: 'inherit',
		fontWeight: '300',
		padding: '.5rem 2.5rem',
		transition: 'all ease-in-out .3s',

		':hover': {
			boxShadow: `0 0 3rem ${RadiumVars.color.colorPrimaryBlue}`
		}
	}
};

export default Radium(Button);
