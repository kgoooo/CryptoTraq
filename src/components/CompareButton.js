import React from 'react';

const CompareButton = (props) => {
	return (
		<div  className="btn--compare">
			<button className="btn btn--light" onClick={props.exchangeSearch}>Track</button>
		</div>
	)
}

export default CompareButton;