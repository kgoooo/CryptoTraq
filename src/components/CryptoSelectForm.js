/*  I refactored this to be a sub component for the new SelectionForm component.
		Once again I realized that to keep the code DRY I needed to refactor the selection
		components to make them more reusable.  This now gets passed into Selection form
		to render the proper content for the select form elements.
		---The main reason why there is still this component and the IntlSelectForm is
		because I couldn't get the crypto.name to render conditionally while being mapped.
*/

import React from 'react';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';


const CryptoSelectForm = (props) =>{
	return (
		<select style={styles.selectForm__form}>
			{props.list.map(crypto => (
				<option key={crypto.code} value={crypto.code}>{crypto.name} â€“ {crypto.code}</option>
			))}
		</select>
	)
};

// Radium styles

const styles = {
	selectForm__form: {
		backgroundColor: RadiumVars.color.colorPrimaryDarkBlue,
		border: `2px solid ${RadiumVars.color.colorPrimaryBlue}`,
		borderRadius: '8px',
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontSize: 'inherit',
		height: '32px',
		outline: 'none',
	},
};


export default Radium(CryptoSelectForm);
