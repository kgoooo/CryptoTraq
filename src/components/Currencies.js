import React from 'react';

const Currencies = (props) => {
	const cryptoList = props.cryptoList;
	const intlList = props.intlList;
	return (
		<div>
      <div>
				<ul>{cryptoList.map(c => <li key={c.code}>{c.name} - {c.code}</li>)}</ul>
        <ul>{intlList.map(i => <li key={i.code}>{i.name}</li>)}</ul>
      </div>
		</div>
	)
}

export default Currencies;