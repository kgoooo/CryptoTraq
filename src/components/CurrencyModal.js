import React, { Component } from 'react';
import ReactModal from 'react-modal';
import SupportedCurrencies from './SupportedCurrencies';

class CurrencyModal extends Component {
  constructor(props) {
		super(props);
		
		this.state = { isOpen: false };
  }
  handleOpenModal = () => {
    this.setState({ isOpen: true });
	}
	
	handleCloseModal = () => {
    this.setState({ isOpen: false });		
	}
  render() {
    let isOpen = this.props.isOpen;
    return (
      <div>
				<SupportedCurrencies modalOpen={this.handleOpenModal} />
				<ReactModal 
					isOpen={this.state.isOpen}
					contentLabel="Modal showing available currencies supported by this app."
					className="Modal"
					overlayClassName="Overlay"
				>
					<button className="btn btn--light" onClick={this.handleCloseModal}>X</button>
				</ReactModal>
      </div>
    );
  }
}

export default CurrencyModal;