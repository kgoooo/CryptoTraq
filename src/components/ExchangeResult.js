/*
Added Radium to this component, nothing else changed.
 */

import React from 'react';
import PropTypes from 'prop-types'
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';

const ExchangeResult = (props) => {
	let showRate;
	if(props.rate !== 0){
		showRate = <h3 style={styles.rate__text}>The exchange rate per {props.crypto} is <span style={styles.rate__span}>{props.rate > 1 ? Math.round(props.rate) : props.rate}</span> {props.intl}</h3>
	}
	if(props.excError === true) {
		showRate = <h3 style={styles.rate__text}>The exchange rate could not be calculated for these currencies, please try another option.</h3>
	}
	return(
		<div>
			{showRate}
		</div>
	)
};

ExchangeResult.propTypes = {
	crypto: PropTypes.string.isRequired,
	intl: PropTypes.string.isRequired,
	rate: PropTypes.number.isRequired,
	excError: PropTypes.bool.isRequired
};

//  Radium Styles below

const styles = {
	rate__text: {
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontWeight: '300',
		fontSize: '24px',
		textAlign: 'center',
		marginBottom: '30px',

		'@media (max-width: 500px)': {
			padding: '20px',
			textAlign: 'center'
		}
	},
	rate__span: {
		fontWeight: '500'
	}
};

export default Radium(ExchangeResult);
export { ExchangeResult }