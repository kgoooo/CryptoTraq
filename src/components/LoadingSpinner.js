/*
		Created this component to display a loading spinner when the user starts
		to track a currency pair.  It shows for both success cases and failures in case
		a user is on a slower connection they still know that their data is being fetched.
		It is reusable and has its information passed in via props.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium'
import {BarLoader} from 'react-spinners';
import RadiumVars from '../RadiumVariables';

const LoadingSpinner = (props) => {
	return (
		<div style={styles.loading__spinner}>
			<BarLoader
				loaderStyle={{display: "block", margin: "0 auto"}}
				color={RadiumVars.color.colorPrimaryLightBlue}
				loading={props.loading}/>
			<p style={styles.loading__text}>{props.content}</p>
		</div>

	)
};

LoadingSpinner.propTypes = {
	loading: PropTypes.bool.isRequired,
	content: PropTypes.string
};

const styles = {
	loading__spinner: {
		display: 'block',
		margin: "30px auto 0 auto"
	},
	loading__text: {
		textAlign: 'center'
	}
};

export default Radium(LoadingSpinner);
export { LoadingSpinner };