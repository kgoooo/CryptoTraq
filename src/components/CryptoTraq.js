/*
		Did some cleaning up of imports in this component and added Radium for the arrow
		styling.
		This component had some refactoring when I made the SelectionForm component to
		break up the big form component I had before.  Mainly adding the new components
		and changing the props that are passed in accordingly.
		Button component is also used here now instead of CompareButton.
 */
/*
	I added two loading spinner components to be rendered here.  One will show up when the app
	is initializing, especially if the user is on a slower connection.  The second will show
	when the user has tracked currencies and is waiting for the results.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import Button from './Button';
import SelectionForm from './SelectionForm';
import CryptoSelectForm from './CryptoSelectForm';
import IntlSelectForm from './IntlSelectForm';
import ExchangeResult from './ExchangeResult';
import LoadingSpinner from './LoadingSpinner';
import Graph from './Graph';
import GraphOptionsBar from './GraphOptionsBar';
import RadiumVars from '../RadiumVariables';

const CryptoTraq = (props) => {
  return(
    <div>
        <div>
          <SelectionForm
              formLabel={"Select Cryptocurency:"}
              selectionFunc={props.handleCryptoChange}
              selectForm={<CryptoSelectForm list={props.cryptoList} delay={styles.delay1} anim={styles.select__glow}/>}
          />
          <p style={styles.arrow}>&rarr;</p>
          <SelectionForm
            formLabel={"Select international currency:"}
            selectionFunc={props.handleIntlChange}
						selectForm={<IntlSelectForm list={props.intlList} delay={styles.delay2} anim={styles.select__glow}/>}
					/>
          <Button onClick={props.handleExchangeSearch} buttonContent={"Track currency"} buttonAnimation={styles.trackAnim}/>
        </div>
				<div>
					<GraphOptionsBar handleRangeZeroChange={props.handleRangeZeroChange} rangeContent={props.rangeContent}/>
				</div>
				{props.initializing ?
					<LoadingSpinner
						loading={props.initializing}
						content={"Loading currencies"}
					/> : null}
				{props.loading ?
					<LoadingSpinner
						loading={props.loading}
						content={'Fetching your results, please hold on.'}
					/>: null}
        {props.showingExchangeRate === true ? (
          <ExchangeResult
            crypto={props.selectedCrypto}
            intl={props.selectedIntl}
            rate={props.exchangeRate}
            excError={props.showingExchangeError}
          />
        ) : null}
        <Graph
          cryptoData={props.dataSet}
          crypto={props.selectedCrypto}
					rangeZero={props.rangeZero}
        />
			</div>
  )
};

CryptoTraq.propTypes = {
	cryptoList: PropTypes.array.isRequired,
	dataSet: PropTypes.array.isRequired,
	exchangeRate: PropTypes.number.isRequired,
	initializing: PropTypes.bool.isRequired,
	intlList: PropTypes.array.isRequired,
	loading: PropTypes.bool.isRequired,
	rangeContent: PropTypes.string.isRequired,
	rangeZero: PropTypes.bool.isRequired,
	selectedCrypto: PropTypes.string,
	selectedIntl: PropTypes.string,
	showingExchangeError: PropTypes.bool.isRequired,
	showingExchangeRate: PropTypes.bool.isRequired,
	handleCryptoChange: PropTypes.func.isRequired,
	handleIntlChange: PropTypes.func.isRequired,
	handleExchangeSearch: PropTypes.func.isRequired,
};

//  Radium Styles
/*
		Added selectGlowAnim to add a small animation to the selection element to indicate to the user
		that they should take action there first.  Applied with select__glow
		I added this trackBtnAnimation to add a small scale animation to the track button as the
		3rd step to tracking your conversion rate.  It gets passed in as props and set in the button component
 */

const selectGlowAnim = Radium.keyframes({
	'50%': {boxShadow: `0 0 3px 6px ${RadiumVars.color.colorPrimaryLightBlue}`},
});
const trackBtnAnimation = Radium.keyframes({
	'0': {transform: 'scale(1)'},
	'50%': {
		transform: 'scale(1.1)',
		boxShadow: `0 0 3px 6px ${RadiumVars.color.colorPrimaryLightBlue}`
	},
	'100%': {transform: 'scale(1)'}
}, 'track');

const styles = {
	arrow: {
		display: 'inline-block',
		fontSize: '50px',
		color: RadiumVars.color.colorPrimaryBlue,
		textShadow: `0px 0px 25px ${RadiumVars.color.colorPrimaryBlue}`,

		'@media (max-width: 500px)': {
			margin: '0px'
		}
	},
	select__glow: {
		animationName: selectGlowAnim,
		animationDuration: '1.5s',
	},
	trackAnim: {
		animationName: trackBtnAnimation,
		animationDuration: '1.5s',
		animationDelay: '3.5s'
	},
	delay1: { animationDelay: '0.85s' },
	delay2: { animationDelay: '2.35s'}
};

export default Radium(CryptoTraq);