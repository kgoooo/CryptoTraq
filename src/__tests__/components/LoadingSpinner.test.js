import React from 'react';
import { shallow } from 'enzyme';
import { LoadingSpinner } from '../../components/LoadingSpinner';


describe('<LoadingSpinner />', () => {

	it('should render correctly', () => {
		const wrapper = shallow(<LoadingSpinner loading={true} content={"Test Spinner"}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('should render no spinner if loading FALSE', () => {
		const wrapper = shallow(<LoadingSpinner loading={false} content={"Test Spinner"}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('should display the correct text content', () => {
		const wrapper = shallow(<LoadingSpinner loading={true} content={"Test Spinner"}/>);
		const paragraphEl = wrapper.find('p');
		expect(paragraphEl.text()).toEqual("Test Spinner");
	});
});