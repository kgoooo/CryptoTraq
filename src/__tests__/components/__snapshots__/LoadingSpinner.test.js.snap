// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> should render correctly 1`] = `
<div
  style={
    Object {
      "display": "block",
      "margin": "30px auto 0 auto",
    }
  }
>
  <onlyUpdateForKeys(Loader)
    color="#00FFF5"
    height={4}
    heightUnit="px"
    loaderStyle={
      Object {
        "display": "block",
        "margin": "0 auto",
      }
    }
    loading={true}
    width={100}
    widthUnit="px"
  />
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Test Spinner
  </p>
</div>
`;

exports[`<LoadingSpinner /> should render no spinner if loading FALSE 1`] = `
<div
  style={
    Object {
      "display": "block",
      "margin": "30px auto 0 auto",
    }
  }
>
  <onlyUpdateForKeys(Loader)
    color="#00FFF5"
    height={4}
    heightUnit="px"
    loaderStyle={
      Object {
        "display": "block",
        "margin": "0 auto",
      }
    }
    loading={false}
    width={100}
    widthUnit="px"
  />
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Test Spinner
  </p>
</div>
`;
